src/list.rs

```diff
@@ -141,4 +133,57 @@ impl<T: Clone> List<T> {
             },
         }
     }
+
+    pub fn sort(&self, cmpfn: fn(T, T) -> i8) -> Self {
```

Comment:
We should use `std::cmp::Ordering` instead of i8

Reply:
Done